src/TcUnit.TestAdapter/Models/SystemService.cs

```diff
@@ -300,13 +318,16 @@ public bool DeleteDirectoryContentRecursive(string path, string osName)
             foreach (AdsFileSystemEntry f in files)
             {
                 string filepath = path + seperator + f.FileName;
+
                 var ret = true;
                 if (!IsValidFilename(f.FileName))
                     ;
                 else if (f.fileAttributes.Directory)
                 {
                     ret = DeleteDirectoryContentRecursive(filepath, seperator);
-                    ret = AdsFileSystem.RemoveDirectory(target, port, filepath, AdsDirectory.PATH_GENERIC);
+                    // for some reason deleting the Plc folder fails, so skip it
+                    if (f.FileName != "Plc")
```

Comment:
On my system, the Plc folder would become locked and it was not possible (via remote desktop) to open the folder or delete it until the TwinCAT system service was restarted. I think the ADS client keeps a lock on the folder even after the connection is closed. This would then prevent the copying to the folder during the next run.